AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/SubscriptionInterceptor/IAMAuthInterceptor.swift

```diff
@@ -94,61 +113,82 @@ class IAMAuthInterceptor: AuthInterceptor {
             return nil
         }
         semaphore.wait()
+
         let authorization = mutableRequest.allHTTPHeaderFields?[SubscriptionConstants.authorizationkey] ?? ""
         let securityToken = mutableRequest.allHTTPHeaderFields?[RealtimeProviderConstants.iamSecurityTokenKey] ?? ""
-        let authHeader = IAMAuthenticationHeader(authorization: authorization,
-                                                 host: host,
-                                                 token: securityToken,
-                                                 date: date,
-                                                 accept: RealtimeProviderConstants.iamAccept,
-                                                 contentEncoding: RealtimeProviderConstants.iamEncoding,
-                                                 contentType: RealtimeProviderConstants.iamConentType)
-        return authHeader
+        let remainingHeaders = mutableRequest.allHTTPHeaderFields?.filter { (header) -> Bool in
+            return !Self.lowercasedHeaderKeys.contains(header.key.lowercased())
+        }
+
+        return IAMAuthenticationHeader(host: host,
+                                       authorization: authorization,
+                                       securityToken: securityToken,
+                                       amzDate: amzDate,
+                                       accept: RealtimeProviderConstants.iamAccept,
+                                       contentEncoding: RealtimeProviderConstants.iamEncoding,
+                                       contentType: RealtimeProviderConstants.iamConentType,
+                                       remainingHeaders: remainingHeaders)
     }
 }
 
 /// Authentication header for IAM based auth
-private class IAMAuthenticationHeader: AuthenticationHeader {
+class IAMAuthenticationHeader: AuthenticationHeader {
     let authorization: String
     let securityToken: String
-    let date: String
+    let amzDate: String
     let accept: String
     let contentEncoding: String
     let contentType: String
+    let remainingHeaders: [String: String]?
```

Comment:
nit: naming, maybe `additionalHeaders`/`extraHeaders`/`customHeaders` would be more appropriate? I'd also add some documentation around this new parameter